helios-ts/example/src/worker/worker.ts

```diff
@@ -0,0 +1,115 @@
+import {createHeliosProvider} from '../../../dist/lib';
+
+import type { Config as NetworkConfig, HeliosProvider, Request, Config, NetworkKind } from '../../../dist/lib'
+
+type IndexedNetworks = {
+  [key: string]: {
+    cfg: NetworkConfig,
+    kind: NetworkKind,
+    provider?: HeliosProvider | null
+  };
+};
+let networks: IndexedNetworks = {}
+
+self.onmessage = async (e) => {  
+    let res  
+    switch (e.data.method) {
+        case 'networks':
+            res = await handleNetworks(e.data.params);            
+            self.postMessage({jsonrpc: '2.0', id: e.data.id, result: res})
+            break;
+        case 'eth_rpc_req':
+            res = await handleEthRpcReq(e.data.params)            
+            self.postMessage({jsonrpc: '2.0', id: e.data.id, result: res})
+            break;
+        default:
+            console.error('Unhandled message/<method>:', e.data.method);
+    }
+};
+
+const handleNetworks = async (options: {method: string, params: {name?: string, cfg: Config, kind: NetworkKind}}): Promise<any> => {
+    const { method, params: { name = '', cfg, kind } } = options; // Set default value for name
+    // const method = options.method
+    // const params = options.params
+    
+    return new Promise(async (resolve, reject) => {
```

Comment:
Here and in other instances of the PR you're using an `async` function -- that always returns a promise -- along with explicitly returning a `Promise`. While it's not a mistake, a cleaner approach would be to just stick to `async`, returning value and `throw`ing in try/catch block